rasa/shared/core/events.py

```diff
@@ -489,17 +489,32 @@ def _entity_string(self):
         return ""
 
     def as_story_string(self, e2e: bool = False) -> Text:
+        """Return event as string for Markdown training format.
+
+        Args:
+            e2e: `True` if the the event should be printed in the format for
+                end-to-end conversation tests.
+
+        Returns:
+            Event as string.
+        """
+        if self.use_text_for_featurization:
```

Comment:
@Ghostvv I think printing an end-to-end `UserUttered` event in the conversation test format will just make things worse and more complicated. I'd vote to fail early. What do you think? 
I checked the usage and we should be fine except some stuff for the story validation tool which for some reason uses this.

Reply:
is it used only in markdown, then ok

Reply:
I'd prefer yaml writer to also use method inside the event then everything is in one place

Reply:
Joe and I just had quite a session about it and the fact that `as_story_string` is used for varying use cases (fingerprinting, printing stories, story validation) actually makes changing code quite complicated. I prefer having the representation of an event in a specific format separated out. How an event is featurized / unique represent should be part of the `Event` , but it shouldn't be related to any markdown stuff.